invoice/models.py

```diff
@@ -1,22 +1,26 @@
 from enum import IntEnum
 
+from django.conf import settings
 from django.db import models
 from django.contrib.contenttypes.fields import GenericForeignKey
 from django.contrib.contenttypes.models import ContentType
-from core.models import HistoryBusinessModel, HistoryModel
+from graphql import ResolveInfo
+
+from core.models import HistoryBusinessModel, HistoryModel, UUIDModel, ObjectMutation, MutationLog
 from core.fields import DateTimeField, DateField
 from datetime import date
 from jsonfallback.fields import FallbackJSONField
 from invoice.apps import InvoicePaymentConfig
 from django.utils.translation import gettext as _
 # Create your models here.
+from invoice.mixins import GenericInvoiceQuerysetMixin, GenericInvoiceManager
 
 
 def get_default_currency():
     return InvoicePaymentConfig.default_currency_code
 
 
-class Invoice(HistoryBusinessModel):
+class Invoice(GenericInvoiceQuerysetMixin, HistoryBusinessModel):
```

Comment:
as you add the mixin, can you change the recipient to thirdparty or you want to do another PR for that ?

Reply:
Can you elaborate, is the idea to create a mixin that would allow to use queries generically depending on whether invoice or bill is considered? if that's the case I think it'd be better to create separate PR, since this one is already quite big.